lib/Bio/EnsEMBL/DataCheck/Checks/MemberProductionCounts.pm

```diff
@@ -0,0 +1,204 @@
+      FROM gene_member_hom_stats 
+    WHERE collection = ?
+  /;
+
+  my $sqlCollections = qq/
+     SELECT DISTINCT clusterset_id 
+       FROM gene_tree_root 
+     WHERE $constraint
+   /;
+
+  my $collections = $helper->execute_simple( -SQL => $sqlCollections );
+
+  my $sqlFamilies = q/
+    SELECT COUNT(*) 
+      FROM family
+  /; # To be tested only with default
+  my $family_count = $helper->execute_single_result( -SQL => $sqlFamilies );
+
+  my $sqlPolyploids = q/
+    SELECT COUNT(*) 
+      FROM genome_db 
+    WHERE genome_component IS NOT NULL
+  /; # Assumes that the polyploid genomes are found in all the collections
+  my $polyploid_count = $helper->execute_single_result( -SQL => $sqlPolyploids );
+  
+  my $sqlGeneTrees = qq/
+    SELECT COUNT(*) 
+      FROM gene_tree_root 
+    WHERE clusterset_id = ?
+  /;
+  
+  my $sqlCAFETrees = qq/
+    SELECT COUNT(*) 
+      FROM gene_tree_root 
+        JOIN CAFE_gene_family 
+          ON gene_tree_root.root_id = gene_tree_root_id 
+    WHERE clusterset_id = ?
+  /;
+  
+  my @columns = qw(families gene_trees gene_gain_loss_trees orthologues paralogues homoeologues);
+
+  foreach my $collection ( @$collections ) {
+
+    my $sums = $helper->execute( -SQL => $sql_sums, -USE_HASHREFS => 1, PARAMS => [$collection] );
+
+    my $genetree_count = $helper->execute_single_result( -SQL => $sqlGeneTrees,  PARAMS => [$collection] );
+    my $cafetrees_count = $helper->execute_single_result( -SQL => $sqlCAFETrees, PARAMS => [$collection] );
+    
+    my @counts = ($family_count, $genetree_count, $cafetrees_count, $genetree_count, $genetree_count, $genetree_count, $polyploid_count);
+
+    if ( $dba->get_division() =~ /vertebrates/ && $collection =~ /default/ ) {
+      my $desc_5 = "The sum of entries for families in gene_member_hom_stats > 0 for the $collection collection";
+      cmp_ok( $sums->[0]->{sum_families}, ">", 0, $desc_5 );
+    }
+    if ( $dba->get_division() =~ /vertebrates/ || $collection =~ /default/ ) {
+      my $desc_6 = "The sum of entries for gene_trees in gene_member_hom_stats > 0 for the $collection collection";
+      cmp_ok( $sums->[0]->{sum_gene_trees}, ">", 0, $desc_6 );
+    }
+
+    foreach ( my $i = 0; $i < @columns; $i++ ) {
+      my $desc_7 = "Found expected entries in gene_member_hom_stats with $columns[$i] > 0 for the $collection collection";
+      my $desc_8 = "There were no unexpected entries in gene_member_hom_stats with $columns[$i] > 0 for the $collection collection";
+      next if ( $collection !~ /default/ && $columns[$i] eq "families" );
+      next if ( $dba->get_division() !~ /plants/ && $columns[$i] eq "homoeologues" );
+      my $col_name = "sum_$columns[$i]";
+
+      if ( $sums->[0]->{$col_name} > 0 ) {
+        ok( $counts[$i] != 0, $desc_7 );
+      }
+      elsif ( $sums->[0]->{$col_name} == 0 ) {
+        ok( $counts[$i] < 1, $desc_8 );
+      }
```

Comment:
Since `< 1` is the same as `== 0` and `> 0` is the same as `!= 0`, you could combine these into a single test:
```
is($sums->[0]->{$col_name} > 0, $counts->[$i] > 0);
```

And for the record, although I find it less readable, there is a Perl habit of using two not `!` operators in a row to normalize expressions to booleans
```
is(!!$sums->[0]->{$col_name}, !!$counts->[$i]);
```